lib/logging/sink.js

```diff
@@ -0,0 +1,214 @@
+     * Get the sink's metadata.
+     *
+     * @resource [Sink Resource]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks#LogSink}
+     * @resource [projects.sink.get API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks/get}
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.metadata - The sink's metadata.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * sink.getMetadata(function(err, metadata, apiResponse) {});
+     */
+    getMetadata: true
+  };
+
+  ServiceObject.call(this, {
+    parent: logging,
+    baseUrl: '/sinks',
+    id: name,
+    createMethod: logging.createSink.bind(logging),
+    methods: methods
+  });
+
+  this.name = name;
+}
+
+nodeutil.inherits(Sink, ServiceObject);
+
+/**
+ * Set the sink's filter.
+ *
+ * This will override any filter that was previously set.
+ *
+ * **This method only works if you are authenticated as yourself, e.g. using the
+ * gcloud SDK.**
+ *
+ * @resource [Advanced Logs Filters]{@link https://cloud.google.com/logging/docs/view/advanced_filters}
+ *
+ * @param {string} filter - The new filter.
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * sink.setFilter('metadata.severity = ALERT', function(err, apiResponse) {});
+ */
+Sink.prototype.setFilter = function(filter, callback) {
+  this.setMetadata({
+    filter: filter
+  }, callback);
+};
+
+/**
+ * Set the sink's metadata.
+ *
+ * **This method only works if you are authenticated as yourself, e.g. using the
+ * gcloud SDK.**
+ *
+ * @resource [Sink Resource]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks#LogSink}
+ * @resource [projects.sink.update API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks/update}
+ *
+ * @param {object} metadata - See a
+ *     [Sink resource](https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks#LogSink).
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var metadata = {
+ *   filter: 'metadata.severity = ALERT'
+ * };
+ *
+ * sink.setMetadata(metadata, function(err, apiResponse) {});
+ */
+Sink.prototype.setMetadata = function(metadata, callback) {
+  var self = this;
+
+  callback = callback || util.noop;
+
+  this.getMetadata(function(err, currentMetadata, apiResponse) {
+    if (err) {
+      callback(err, apiResponse);
+      return;
+    }
+
+    self.request({
+      uri: '',
```

Comment:
Our library patches `request`s upwards through the inheritance model. `request.uri` in this context will resolve to `https://logging.googleapis.com/v1beta3/projects/{projectId}/sinks/{thisSinkName}`. Anything specified in `uri` would get appended to that base uri.